server/src/Application/Services/FlashcardImporter/FlashcardImporter.cs

```diff
@@ -22,21 +22,39 @@ public async Task<IReadOnlyCollection<CreateFlashcardDto>> ImportFlashcardsFromC
 
         var lines = stream.ReadLines();
         var importedFlashcards = new BlockingCollection<CreateFlashcardDto>();
+        var threadList = new List<Thread>();
+        var lockObject = new object(); 
 
-        Parallel.ForEach(lines, line =>
+        foreach (var line in lines)
         {
-            var values = line.Split(',');
-            if (values.Length != 2)
-            {
-                throw new ValidationException($"CSV file must only contain flashcard front and back values. Invalid values: {string.Join(',', values)}");
-            }
-
-            var front = values[0];
-            var back = values[1];
-            var createFlashcardDto = new CreateFlashcardDto(flashcardSetId, front, back);
+            var localLine = line; 
 
-            importedFlashcards.Add(createFlashcardDto);
-        });
+            var thread = new Thread(() =>
```

Comment:
Each line should not be parsed on separate thread since that is too resource expensive. For example, you wouldn't want to start 100 threads for 100 lines. Probably the array should be divided in multiple batches (e.g. 2 or 3) and start a new thread for those batches.